cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/output/ProvidedOutputFormat.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.batch.dataset.output;
+
+import co.cask.cdap.api.data.batch.OutputFormatProvider;
+
+import java.util.Map;
+
+/**
+ * Internal helper class to represent an output format provider and its configuration.
+ */
+public class ProvidedOutputFormat {
```

Comment:
previously the context was maintaining a map from name/alias to output format provider, now we call getConfig() immediately and therefore also need to remember the configuration. That is the purpose of this class.